### ISSUE TITLE
iOS Release builds stopped working after last update

### ISSUE THREAD

**Issue by AbstractionsAs** [dotnet/maui#12519]
### Description

App crashes immediatly with XamlParseException "Cannot assign property "Property": Property does not exist, or is not assignable, or mismatching type between value and property (XamlParseException)".

This seems to be caused by style setters?
Works fine in debug, also works fine for all builds on Android.

### Steps to Reproduce

It seems like this has to do with style setters beeing removed by the linker. It happens immeiatly when I launch the app. 

### Link to public reproduction project repository

None

### Version with bug

7.0 (current)

### Last version that worked well

7.0 (current)

### Affected platforms

iOS

### Affected platform versions

iOS 16.2

### Relevant log output

```shell
XamlParseException "Cannot assign property "Property": Property does not exist, or is not assignable, or mismatching type between value and property (XamlParseException)".
```

**Comment by AbstractionsAs** [comment]
I'm expremely pressed for time, and I don't know if I can do this in the timeframe required. Meanwhile, I'm deploying debug builds to the AppStore, while looking into a way to use an older version of Maui in my iOS pipeline.

There's a number of serious bugs I've simply skipped reporting because of this open-repo requirement. It's too much work I don't have time for, and it doesn't really feel like the team takes bug reports seriously anyway.

Maui is not production-ready (unfortunaly for me, given that I'm developing a major project using it).

**Comment by breenbob** [comment]
Actually it seems disabling linker on iOS Release builds does not resolve this issue, making this is a much more serious issue. But positively removing our DataTrigger does prevent the crash, so hopefully a pointer in what to test for a repro and fix.

**Comment by rolfbjarne** [comment]
One idea might be to enable the interpreter on a Release build, and see if that changes anything:

```xml
<PropertyGroup>
    <UseInterpreter>true</UseInterpreter>
</PropertyGroup>
```

**Comment by breenbob** [comment]
Thanks @rolfbjarne. Will give it a go and add an update.

**Comment by minglu** [comment]
Same release build issue, UserInterpreter true doesn't work for me, I have to select [Don't Link] to make it work, but we couldn't upload it to TestFlight with [Don't Link], it would be rejected by App store with a long list of missing purpose string errors 

`ITMS-90683: Missing purpose string in Info.plist`
`ITMS-90078: Missing Push Notification Entitlement`

**Comment by rolfbjarne** [comment]
@minglu 

> UserInterpreter true doesn't work for me

Note that it's `UseInterpreter` (not  `UserInterpreter`).

**Comment by minglu** [comment]
@rolfbjarne Thanks, sorry for the typo

```xml
        <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-ios|AnyCPU'">
		<CreatePackage>false</CreatePackage>
		<EnableAssemblyILStripping>false</EnableAssemblyILStripping>
		<ArchiveOnBuild>true</ArchiveOnBuild>
		<RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
		<CodesignProvision>Automatic</CodesignProvision>
		<CodesignKey>iPhone Distribution</CodesignKey>
	</PropertyGroup>
	<PropertyGroup>
		<UseInterpreter Condition="$(TargetFramework.EndsWith('-ios'))">true</UseInterpreter>
	</PropertyGroup>```

**Comment by rolfbjarne** [comment]
@minglu and even then you get exactly this exception like the original reporter?

> XamlParseException "Cannot assign property "Property": Property does not exist, or is not assignable, or mismatching type between value and property (XamlParseException)".

If so, would you be able to create a test project we can use to reproduce this?

**Comment by minglu** [comment]
@rolfbjarne Thanks for the reply, actually mine is not getting this exact error
it's #8122, or [78889 from dotnet runtime](https://github.com/dotnet/runtime/issues/78889)
during deserializing with the iOS release build, I can create a new issue to avoid any confusion

**Comment by rolfbjarne** [comment]
> @rolfbjarne Thanks for the reply, actually mine is not getting this exact error
> it's #8122, or [75802 from dotnet runtime](https://github.com/dotnet/runtime/issues/75802) and [78889 from dotnet runtime](https://github.com/dotnet/runtime/issues/78889)
> during deserializing with the iOS release build, I can create a new issue to avoid any confusion

Yes, please file a new issue.

**Comment by RobTF** [comment]
Hi, I'm getting this exact error on iOS release build;

```
*** Terminating app due to uncaught exception 'Microsoft.Maui.Controls.Xaml.XamlParseException', reason: 'Position 7:13. Cannot assign property "Property": Property does not exist, or is not assignable, or mismatching type between value and property (Microsoft.Maui.Controls.Xaml.XamlParseException)
   at Microsoft.Maui.Controls.Xaml.ApplyPropertiesVisitor.SetPropertyValue(Object xamlelement, XmlName propertyName, Object value, Object rootElement, INode node, HydrationContext context, IXmlLineInfo lineInfo)
   at Microsoft.Maui.Controls.Xaml.ApplyPropertiesVisitor.Visit(ValueNode node, INode parentNode)
   at Microsoft.Maui.Controls.Xaml.ValueNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Microsoft.Maui.Controls.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Microsoft.Maui.Controls.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Microsoft.Maui.Controls.Xaml.FillResourceDictionariesVisitor.Visit(ElementNode node, INode parentNode)
   a<…>
```

It's odd that it claims I'm setting a property named `Property` as that doesn't appear to be the case anywhere in the XAML.

Any tips on debugging this? This message is all I get in the iOS device logs, and doesn't mention a XAML filename etc or give me any other thread to pull at it seems.

I've ported over a medium sized app, and whilst we have Android up and running, this is the last major hurdle before we can push to Testflight for a wider scoped acceptance test - any help appreciated!

**Comment by rolfbjarne** [comment]
@RobTF did you try enabling the interpreter: https://github.com/dotnet/maui/issues/12519#issuecomment-1409099527?

**Comment by RobTF** [comment]
Trying it now - just waiting for it to process in the app store.

Quick question - to make the debug/dev loop tighter on this - is it possible to deploy a release build directly to a device using VS? I've tried switching the setting to Release and building but it's not having it - could it just be my config?

I'll reply again as soon as I get an answer on the interpreter version.

**Comment by RobTF** [comment]
@rolfbjarne Yep, that works for me! Does that mean it's just a matter of keeping this setting enabled until a bug is fixed on hte MAUI side, or is there something I could be getting wrong?

thanks

**Comment by rolfbjarne** [comment]
@RobTF if you can provide a test project we can use to reproduce this, we might be able to determine if it's a bug that needs to be fixed or not. In any case you can keep this setting at enabled if you need to.

**Comment by RobTF** [comment]
Sure, I can maybe give it a go in my free time.

I'll see if I can break it in release mode locally first so I can get a reasonable dev loop together as I suspect it'll take a fair few iterations, therefore won't be realistic for me to keep throwing builds up to the app store.

**Comment by RobTF** [comment]
Hi,

Please don't close this yet. Despite the compiler flags I'm getting soarodic reports of this crash happening in prod on some iOS devices. Im waiting for details as to the specific devices and OS level affected.

Building a repro for this is extremely hard as so far it clearly only manifests with a non trivial app and requires a full loop through the Apple App store to push out a release build and test it. With a full time job and pressure to get work done finding the time to help fix the MS dev tools is a hard sell internally.

There are loads more areas to dig here anyway, for example what gave @rolfbjarne the idea to enable the interpreter in the first place? That smells of something known about, what does that interpreter flag do specifically? I've read the documentation and get the gist, but still can't grep the finer details. More info would help guide us non MAUI developers towards helping build a targeted repro.

Further, despite this is any effort being put towards hardening the release process on the MAUI side? The release build process seems to suffer from this "last minute bugs" issue with steps such as the linking etc. and other processes which can change the meaning of the tested program at the last minute. Is there an easy way we can replicate these steps with a debugger attached so we can build a repro reasonably easily and be more assured as to the quality of the final binaries? Have the MAUI team gone through the release cycle on iOS with a non trivial app that itself could be used as the base of a repo-case possibly?

Repo or not, I think this issue deserves further conversation. If it helps I'm open to doing a screen share for paired debugging or even supplying my codebase to the MAUI team (under NDA) to be used as an example. What I will struggle with is writing a brand new, non trivial demo app example in my spare time, effectively blind as to what could be causing the issue.

**Comment by PureWeen** [comment]
@RobTF do you have any crash logs generated from the store?

**Comment by rolfbjarne** [comment]
> Building a repro for this is extremely hard as so far it clearly only manifests with a non trivial app and requires a full loop through the Apple App store to push out a release build and test it.

You should be able to install a release build locally by using the same codesigning identity + provisioning profile you use for debug builds.

> There are loads more areas to dig here anyway, for example what gave @rolfbjarne the idea to enable the interpreter in the first place?

Similar-sounding issues with exceptions during release builds (and the knowledge that by default the interpreter is enabled for debug builds in MAUI, but disabled for release builds).

> That smells of something known about, what does that interpreter flag do specifically? I've read the documentation and get the gist, but still can't grep the finer details. More info would help guide us non MAUI developers towards helping build a targeted repro.

On iOS, it's not possible to generate executable machine code at runtime (i.e. JIT compile code), because Apple doesn't allow it. Thus we use an Ahead-Of-Time compiler, which compiles everything to native code at build time - but this has a few limitations (there's no support for System.Reflection.Emit for instance). We developed an interpreter to get around these limitations, and incidentally the interpreter is also required for a few debug features (hot reload for instance), and it's thus enabled by default for debug builds. The downside is that it's fairly easy to end up with a dependency on the runtime features of the interpreter - because it just works during debug.

Why aren't we enabling the interpreter for all build configurations? Because there may be performance concerns in some cases, but we're investigating this and may end up doing exactly that (enabling the interpreter for all build configurations).

> and be more assured as to the quality of the final binaries?

The best way to do this is to create a Release build, but with the codesigning identity + provisioning profile you use for a debug build, and install and test it on a device. You won't be able to attach a debugger, but you'll be able to verify that a release build works as expected.

**Comment by fgiacomelli** [comment]
related to #12316 , not resolved at this time
